src/main/java/de/jojii/matrixclientserver/Bot/Events/RoomEvent.java

```diff
@@ -21,78 +23,56 @@ public RoomEvent(JSONObject raw, String type, String event_id, String sender, St
         this.content = content;
     }
 
-    public JSONObject getRaw() {
-        return raw;
-    }
-
-    public JSONObject getContent() {
-        return content;
-    }
-
-    public String getType() {
-        return type;
-    }
-
-    public String getEvent_id() {
-        return event_id;
-    }
-
-    public String getSender() {
-        return sender;
-    }
-
-    public String getRoom_id() {
-        return room_id;
-    }
-
-    public static String getBodyFromMessageEvent(RoomEvent event){
+    public static String getBodyFromMessageEvent(RoomEvent event) {
         return event.getContent().getString("body");
     }
 
     public static List<RoomEvent> parseAllEvents(JSONObject object) {
         List<RoomEvent> roomEvents = new ArrayList<>();
-        String[] pointsTOP = {"join", "invite", "leave"};
-        for (String pointTop : pointsTOP) {
-
-            JSONObject object1 = object.getJSONObject("rooms").getJSONObject(pointTop);
-            Iterator<String> keys = object1.keys();
-
-            while (keys.hasNext()) {
-                String key = keys.next();
-                JSONObject room = object1.getJSONObject(key);
-
-                String[] points = {"timeline", "state", "account_data", "ephemeral", "invite_state"};
-                for (String point : points) {
-                    if (!room.has(point)) {
-                        continue;
-                    }
-
-                    JSONArray timeline = room.getJSONObject(point).getJSONArray("events");
-                    for (int i = 0; i < timeline.length(); i++) {
-                        JSONObject event = timeline.getJSONObject(i);
-
-                        roomEvents.add(fetchRoomEvent(event, key));
-
+        for (String pointTop : POINTS_TOP) {
+
+            if (object.has("rooms")) {
+                final JSONObject rooms = object.getJSONObject("rooms");
+
+                if (rooms.has(pointTop)) {
```

Comment:
What do you think about negating those if-statements and use `continue` in order to prevent a lot of nesting scopes?

Reply:
Ok. It makes sense to use `continue` in these cases.

Unfortunately I did not create a branch in my fork before creating this PR. Therefore, a few more changes have found their way into this PR.